### PR TITLE
refactor(swipe): creating AssetAddressRepository and WalletSwipeAddressDelegate.

### PR DESCRIPTION
**Pull request by bchrisarriola**
* Migrated out `didGetSwipeAddresses` from `RootService`.
* Created `AssetAddressRepository` component for retrieving addresses—would like to move latestMultiAddress response (currently in WalletManager) in there as well.
* Removed all direct references to the KeychainItemWrapper and instead reference the new `AssetAddressRepository`

### CODE REVIEW

File: PinEntry/Classes/PEPinEntryController.m [hunk 1]
```diff
@@ -190,8 +190,20 @@ - (void)setupQRCode
...
+
+        AssetType type;
+        switch (assetType) {
+            case LegacyAssetTypeBitcoin:
+                type = AssetTypeBitcoin;
+            case LegacyAssetTypeBitcoinCash:
+                type = AssetTypeBitcoinCash;
+            case LegacyAssetTypeEther:
+                type = AssetTypeEthereum;
+        }
```
Comment (woudini) [inline PinEntry/Classes/PEPinEntryController.m]: Any reason to not have a function converting `LegacyAssetType` to `AssetType`?

**Comment by woudini** [comment]
Not compiling for me at 6cac6638a74c67c5c3f3c80b6859e0887180e109
screen shot 2018-05-22 at 9 00 02 pm

File: PinEntry/Classes/PEPinEntryController.m (hunk 1 above)
Comment (bchrisarriola) [reply PinEntry/Classes/PEPinEntryController.m]: Mostly because it's going to be deprecated. But let me add a method within `PEPinEntryController`.

**Comment by bchrisarriola** [comment]
Hmm try doing a clean build? I'm not getting that issue

**Comment by woudini** [comment]
It compiles now 👍

**Pull request merged**